sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -766,12 +767,47 @@ private Mono<Void> sendIterable(Iterable<ServiceBusMessage> messages, ServiceBus
         if (Objects.isNull(messages)) {
             return monoError(LOGGER, new NullPointerException("'messages' cannot be null."));
         }
+        final Iterator<ServiceBusMessage> messagesItr = messages.iterator();
+        if (messagesItr.hasNext()) {
+            return sendNextIterableBatch(messagesItr.next(), messagesItr, transaction);
+        } else {
+            return Mono.empty();
+        }
+    }
 
-        return createMessageBatch().flatMap(messageBatch -> {
-            StreamSupport.stream(messages.spliterator(), false)
-                .forEach(message -> messageBatch.tryAddMessage(message));
```

Comment:
So, here we used to create one batch and add Iterable messages, but the return value of `tryAddMessage` is ignored resulting those messages to be dropped.

Reply:
Good catch